core/domain/topic_domain.py

```diff
@@ -554,34 +554,6 @@ def create_default_topic(cls, topic_id, name):
             feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION,
             constants.DEFAULT_LANGUAGE_CODE, 0)
 
-    @classmethod
-    def update_subtopics_from_model(
```

Comment:
Why is this deleted? This is a really important function that initiates subtopic migration.
You can't reach this code now as subtopic version is 1. Though, this would be needed in the future when subtopic migrations would be done.
@seanlip What do you think?

Reply:
Hmm I thought we should not have code that is of no use in our codebase -- we can write these functions in the future when they are actually needed. Isn't this what @seanlip said in Oppia's last presentation?

Reply:
We do need to keep this one, since every object that's stored as a json schema needs an upgrade pipeline (and we should have that upgrade pipeline in place before we release even v1 of the object).

Reply:
Then for the time being, I am using self.swap to get this code executed.

Reply:
See if you can find a general pattern for testing these upgrade pipelines that's consistent throughout the codebase.

If you need to, you can make a mock subclass of the domain object class and add a convert_v1_to_v2 method for now, with a TODO to remove that test subclass once there's a real method in the main code to use.

Reply:
Also /cc @aks681 for any suggestions on the best way to test this.

Reply:
@seanlip I think your suggestion is good. @aks681 I followed that suggestion and created a mock class to test the migrations.